File: sav_frontend/src/types/article.ts
// types/article.ts
export interface Article {
  id: number;
  reference: string;
  nom: string;
  type: string;
  description?: string;
  prixAchat: number;
  estEnStock: boolean;
  estSousGarantie: boolean; // Propriété calculée côté backend
  dateAchat: string; // Format: "2024-01-15T00:00:00"
  dureeGarantieMois: number;
  imageUrl?: string;
  // Champs optionnels (selon votre modèle backend étendu)
  numeroSerie?: string;
  dateInstallation?: string;
  lieuInstallation?: string;
  typeInstallation?: string;
  // Propriété calculée (backend)
  finGarantie?: string;
}

export interface ArticleStats {
  Total: number;
  Sanitaire: number;
  Chauffage: number;
  EnStock: number;
  SousGarantie: number;
  PrixMoyen: number;
  PrixMax: number;
  PrixMin: number;
}

File: Service_apres_vente_back/ArticleAPI/Models/Article.cs
using System.ComponentModel.DataAnnotations;

namespace ArticleAPI.Models
{
    public class Article
    {
        public int Id { get; set; }

        [Required]
        [StringLength(50)]
        public string Reference { get; set; } = string.Empty;

        [Required]
        [StringLength(100)]
        public string Nom { get; set; } = string.Empty;

        [Required]
        [StringLength(20)]
        public string Type { get; set; } = string.Empty; // "Sanitaire" ou "Chauffage"

        [Required]
        public DateTime DateAchat { get; set; }

        [Required]
        [Range(0, 120)]
        public int DureeGarantieMois { get; set; }

        [StringLength(500)]
        public string? Description { get; set; }

        [Required]
        [Range(0, 999999.99)]
        public decimal PrixAchat { get; set; }

        [StringLength(300)]
        public string? ImageUrl { get; set; }
        // Ajouter ces propriétés pour le SAV
        [Display(Name = "Numéro de série")]
        [StringLength(100)]
        public string? NumeroSerie { get; set; }

        [Display(Name = "Date d'installation")]
        public DateTime? DateInstallation { get; set; }

        [Display(Name = "Lieu d'installation")]
        [StringLength(200)]
        public string? LieuInstallation { get; set; }

        [Display(Name = "Type d'installation")]
        [StringLength(50)]
        public string? TypeInstallation { get; set; } // "Murale", "Sol", "Encastrée"
        public bool EstEnStock { get; set; } = true;

        // Propriété calculée pour savoir si l'article est sous garantie
        public bool EstSousGarantie
        {
            get
            {
                return DateTime.Now <= DateAchat.AddMonths(DureeGarantieMois);
            }
        }

        // Date de fin de garantie
        public DateTime FinGarantie => DateAchat.AddMonths(DureeGarantieMois);
    }
}